riotfile.py

```diff
@@ -81,14 +94,84 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
     return [version_to_str(version) for version in SUPPORTED_PYTHON_VERSIONS if min_version <= version <= max_version]
 
 
+# Decoration of the Venv Class with a new field 'ci' to store the configuration of each test
+@dataclasses.dataclass
+class Venv(RiotVenv):
+    ci: Dict[str, Any] = None
+
+
+# Use of DD_USE_LATEST_VERSIONS to set PyLatest and use
+# either the fixed versions or the last versions available
+
+LOGGER = logging.getLogger(__name__)
+
+PY_Latest = False
+
+if "DD_USE_LATEST_VERSIONS" not in os.environ:
+    LOGGER.debug("DD_USE_LATEST_VERSIONS not set.")
+elif os.environ["DD_USE_LATEST_VERSIONS"].lower() == "true":
+    LOGGER.debug("Use LATEST versions of packages")
+    PY_Latest = True
+else:
+    LOGGER.debug("Use regular fixed versions of packages")
+
+
+def latest(arg: str):
+    """Compute the latest version, either fixed from LATEST_VERSIONS or latest available"""
+    if PY_Latest:
+        return riot_latest
+    else:
+        try:
+            return "<=" + LATEST_VERSIONS[arg]
```

Comment:
it will not be ok, because pypi uses this contraint to install dependency by taking into account the python version. For some tests, using latest is equivalent for pypi to "use the latest compatible version for this particular version of python". But in the fixed version, if we say '==' or '~=', it will fail for many tests, especially for 2.7 or 3.5